tests/plenary/parser/search_spec.lua

```diff
@@ -54,10 +54,11 @@ describe('Search parser', function()
     assert.Is.False(result:check({ tags = 'OTHER' }))
```

Comment:
Why are changes to this test file needed? I don't see any changes in the `Search` class.

Reply:
Sorry it's been a while since I did this development so I don't remember off hand. I'm trying to check compare my version of `search_spec.lua` to the current master version to see if anything breaks due to the feature I implemented. I'm having trouble getting tests to run though. I'm getting a lot of errors and it's apparently due to missing an orgmode TS grammar. I don't remember seeing this error before and I'm wondering if its related to the addition of the new folding logic.

I'd appreciate any guidance you can provide so I can give you a better answer for why these changes were necessary.

Reply:
No worries. I don't think you introduced any changes that would break anything, you just added new behavior. I would suggest to start with reverting changes to the test files and go from there.
Regarding running tests locally, this should get you covered https://github.com/nvim-orgmode/orgmode#tests . This should work fine for Linux/MacOS. I'm not sure about Windows to be honest.

Reply:
I've followed that procedure in the past without issue. Now I'm seeing the following errors in several test modules that I don't think my code interacts with.

For example, I'm seeing the following output as part of `make test`:
```
Testing:        /home/cflowers/.local/share/nvim/lazy/orgmode/tests/plenary/notifications/notifications_spec.lua
Fail    ||      Notifications should find headlines for notification
            /usr/share/nvim/runtime/lua/vim/treesitter/language.lua:32: no parser for 'org' language, see :help treesitter-parsers
            
            stack traceback:
                /usr/share/nvim/runtime/lua/vim/treesitter/language.lua:32: in function 'require_language'
                /usr/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:35: in function 'new'
                ./lua/orgmode/parser/file.lua:173: in function 'from_content'
                ...gmode/tests/plenary/notifications/notifications_spec.lua:32: in function <...gmode/tests/plenary/notifications/notifications_spec.lua:18>
            
Fail    ||      Notifications should find repeatable and warning deadlines for notification
            /usr/share/nvim/runtime/lua/vim/treesitter/language.lua:32: no parser for 'org' language, see :help treesitter-parsers
            
            stack traceback:
                /usr/share/nvim/runtime/lua/vim/treesitter/language.lua:32: in function 'require_language'
                /usr/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:35: in function 'new'
                ./lua/orgmode/parser/file.lua:173: in function 'from_content'
                ...gmode/tests/plenary/notifications/notifications_spec.lua:98: in function <...gmode/tests/plenary/notifications/notifications_spec.lua:76>
            
Fail    ||      Notifications should allow disabling specific reminder times
            ...gmode/tests/plenary/notifications/notifications_spec.lua:282: attempt to index local 'orgfile' (a nil value)
            
            stack traceback:
                ...gmode/tests/plenary/notifications/notifications_spec.lua:282: in function <...gmode/tests/plenary/notifications/notifications_spec.lua:277>
            
Fail    ||      Notifications should allow disabling specific reminder types
            ...gmode/tests/plenary/notifications/notifications_spec.lua:438: attempt to index local 'orgfile' (a nil value)
            
            stack traceback:
                ...gmode/tests/plenary/notifications/notifications_spec.lua:438: in function <...gmode/tests/plenary/notifications/notifications_spec.lua:433>
            

Success:        0
Failed :        4
Errors :        0
```

The first test mentions that there's no treesitter parse for the `org` language. Has something changed in the `nvim-treesitter` repository to cause this issue? Can you test with a fresh clone to see if it's reproducible?

Reply:
I'm no longer having this issue. It seems it may have been a bug published in Arch's `tree-sitter` package. I'll work on addressing this feedback now.

Reply:
I've appended to this PR with a reversion of the test file back to `master` and then subsequent commits detailing why the minimal changes were necessary. All but one test is currently passing from the suite, but that test is also failing on `master`.